test/addLinkToProjectAction.test.js

```diff
@@ -1,50 +1,43 @@
 const { expect } = require('chai');
 const AddLinkToProjectAction = require('../lib/actions/addLinkToProjectAction');
 const AddLinkToProjectResponse = require('../lib/actions/addLinkToProjectResponse');
+const Project = require('../lib/project')
+const ProjectLinks = require('../lib/projectLinks')
 const AssertionError = require('assert').AssertionError;
 
 describe('Add Link to Project Action', ()=> {
 
     it('returns a string as an object', () => {
         const addLinkToProjectAction = new AddLinkToProjectAction();
-
         const expectedResult = { label: 'google', link: 'google.com' }; 
-
-        expect(addLinkToProjectAction.parseParams('google google.com')).to.deep.equal(expectedResult);
+        expect(addLinkToProjectAction.parseParams('add google google.com')).to.deep.equal(expectedResult);
     })
 
     it('returns an error message on invalid input', () => {
         const addLinkToProjectAction = new AddLinkToProjectAction();
-        
         const parseParamsThatThrows = () => addLinkToProjectAction.parseParams('google')
-
         expect(parseParamsThatThrows).to.throw(TypeError, /invalid/);
     })
 
     it('returns an error message if parameters are undefined', () => {
-        const addLinkToProjectAction = new AddLinkToProjectAction();
-        
+        const addLinkToProjectAction = new AddLinkToProjectAction(); 
         const parseParamsThatThrows = () => addLinkToProjectAction.parseParams()
-
         expect(parseParamsThatThrows).to.throw(TypeError, /invalid/);
     })
 
     it('returns an error message if parameters not a string', () => {
         const addLinkToProjectAction = new AddLinkToProjectAction();
-        
         const parseParamsThatThrows = () => addLinkToProjectAction.parseParams(['google', 'google.com'])
-
         expect(parseParamsThatThrows).to.throw(AssertionError);
     })
 
     xit('@asyncResolves : adds link to DB', async () => {});
 
     it('@asyncResolves : returns an add object containing a response', async () => {
-        const addLinkToProjectAction = new AddLinkToProjectAction();
-
-        const response = await addLinkToProjectAction.execute('google google.com')
-
-        const expectedResult = new AddLinkToProjectResponse('google', 'google.com')
+        const action = new AddLinkToProjectAction();
+        const project = new Project();
+        const response = await action.execute('add google google.com', project);
+        const expectedResult = new AddLinkToProjectResponse('google', 'google.com');
```

Comment:
Let's avoid using a `deep.equal` on the instance of this type. Instead, let's consider something like:

```
expect(typeof response).to.equal(AddLinkToProjectResponse)
```

This allows your `AddLinkToProjectResponse` to have it's own "behavioral" tests while this is just concerned with the response shape.

Reply:
@gtempus what do you think about this? It still feels a little off.